File: app/src/main/java/com/mbcdev/goddamn/GodDamnActivity.java
package com.mbcdev.goddamn;

import android.media.MediaPlayer;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class GodDamnActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_god_damn);

        final MediaPlayer mediaPlayer = MediaPlayer.create(GodDamnActivity.this, R.raw.goddamn);

        findViewById(R.id.goddamn).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (mediaPlayer.isPlaying()) {
                    mediaPlayer.seekTo(0);
                }

                mediaPlayer.start();
            }
        });
    }
}

File: app/src/main/kotlin/goddamn/GodDamnActivity.kt
package goddamn

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.mbcdev.goddamn.R
import kotlinx.android.synthetic.main.activity_god_damn.*

class GodDamnActivity : AppCompatActivity() {

    private val godDamnMessenger by lazy { GodDamnMessenger(this) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_god_damn)

        goddamn.setOnClickListener {
            GodDamnApp.get(this).godDamn()
            godDamnMessenger.shareGodDamn()
        }
    }

    override fun onResume() {
        super.onResume()

        GodDamnApp.get(this).godDamn()
    }

    override fun onStart() {
        super.onStart()
        godDamnMessenger.listenForGodDamns()
    }

    override fun onStop() {
        super.onStop()
        godDamnMessenger.ignoreGodDamns()
    }
}
